instrumentation/google.golang.org/grpc/otelgrpc/interceptor.go

```diff
@@ -208,24 +200,15 @@ func wrapClientStream(s grpc.ClientStream, desc *grpc.StreamDesc) *clientStream
 	go func() {
 		defer close(eventsDone)
 
-		// Both streams have to be closed
-		state := byte(0)
-
 		for event := range events {
 			switch event.Type {
-			case closeEvent:
-				state |= clientClosedState
 			case receiveEndEvent:
-				state |= receiveEndedState
+				finished <- nil
```

Comment:
I was running into the same issues and after applying this PR I was still seeing some leaks.  I think this fixes some of the issues, but I think there is still an edge case where the client context is cancelled in which case we may not receive any more events.  I patched this loop to select on the `s.Context.Done()` along with `events` and all my leaks have gone away.  This is what my loop looks like:
```
		for {
			select {
			case <-s.Context().Done():
				finished <- s.Context().Err()
				return
			case event := <-events:
				switch event.Type {
				case receiveEndEvent:
					finished <- nil
					return
				case errorEvent:
					finished <- event.Err
					return
				}
			}
		}
```

Reply:
Please see https://github.com/open-telemetry/opentelemetry-go-contrib/pull/840